packages/main/src/components/AnalyticalTable/ColumnHeader/index.tsx

```diff
@@ -114,30 +118,46 @@ export const ColumnHeader: FC<ColumnHeaderProps> = (props: ColumnHeaderProps) =>
   } = props;
 
   const isFiltered = column.filterValue && column.filterValue.length > 0;
-  const desc = column.isSortedDesc;
-
-  const sortingIcon = column.isSorted ? <Icon name={desc ? 'sort-descending' : 'sort-ascending'} /> : null;
+  const sortingIcon = column.isSorted ? (
+    <Icon name={column.isSortedDesc ? 'sort-descending' : 'sort-ascending'} />
+  ) : null;
   const filterIcon = isFiltered ? <Icon name="filter" /> : null;
   const groupingIcon = column.isGrouped ? <Icon name="group-2" /> : null;
 
+  const textStyle = useMemo(() => {
+    let margin = 0;
+
+    if (column.isSorted) margin++;
+    if (column.isGrouped) margin++;
+    if (isFiltered) margin++;
+
+    if (margin === 0) {
+      return {};
+    }
+
+    if (margin > 0) margin += 0.5;
+
+    if (getRTL()) {
+      return {
+        marginLeft: `${margin}rem`
+      };
+    }
+    return {
+      marginRight: `${margin}rem`
+    };
+  }, [column.isSorted, column.isGrouped, isFiltered, column.hAlign]);
```

Comment:
Please remove unnecessary `column.hAlign` from dependencies array.
```suggestion
  }, [column.isSorted, column.isGrouped, isFiltered]);
```